Gemfile.lock

```diff
@@ -1,7 +1,7 @@
 PATH
   remote: .
   specs:
-    hocon (1.0.0)
+    hocon (1.0.1)
```

Comment:
The downside of .lock files is that if Gem dependencies change in a way that breaks your code the .lock file will prevent you from seeing the problem. [Other projects explicitly ignore .lock files](https://github.com/puppetlabs/puppetlabs-dsc/blob/master/.gitignore), but like you pointed out there is no official convention for Puppet AFAIK.